task-sdk/src/airflow/sdk/execution_time/context.py

```diff
@@ -180,6 +181,8 @@ def _get_variable(key: str, deserialize_json: bool) -> Any:
                     import json
 
                     var_val = json.loads(var_val)
+                if isinstance(var_val, str):
+                    mask_secret(var_val, key)
```

Comment:
Do we want to also mask e.g. list of str? `mask_secret` can handle that natively.

Reply:
Please feel free to reopen if you disagree, can fix in a follow up :D